src/main/java/org/folio/service/titles/TitlesService.java

```diff
@@ -130,16 +160,175 @@ private List<String> getNonPackageLineIds(List<CompositePoLine> compositePoLines
     return compositePoLines.stream().filter(line -> !line.getIsPackage()).map(CompositePoLine::getId).collect(toList());
   }
 
-  public Future<String> updateTitleWithInstance(String titleId, RequestContext locationContext, RequestContext requestContext) {
+  public Future<String> updateTitleWithInstance(String titleId, RequestContext locationContext,
+                                                RequestContext requestContext) {
     return getTitleById(titleId, requestContext)
       .compose(title -> updateTitleWithInstance(title, false, locationContext, requestContext));
   }
 
-  public Future<String> updateTitleWithInstance(Title title, boolean isInstanceMatchingDisabled, RequestContext locationContext, RequestContext requestContext) {
+  public Future<String> updateTitleWithInstance(Title title, boolean isInstanceMatchingDisabled,
+                                                RequestContext locationContext, RequestContext requestContext) {
     return titleInstanceService.getOrCreateInstance(title, isInstanceMatchingDisabled, locationContext)
       .map(title::withInstanceId)
       .compose(entity -> saveTitle(entity, requestContext)
         .map(v -> entity.getInstanceId()));
   }
 
+  /**
+   * Unlinks a title from a package. This involves checking the existence of holdings and items for the title.
+   * If the holdings have references to other titles, the unlinking process skips them.
+   * If the holdings do not have references, it asks for confirmation to delete related pieces, items, and holdings.
```

Comment:
From what I read in [UIOR-968](https://folio-org.atlassian.net/browse/UIOR-968), if there are no references to other titles, pieces holdings and titles should automatically be deleted by backend (confirmation is done beforehand by UI).

Reply:
5th AC is saying that if holdings don't have reference and then confirmation model should be displayed, so currently logic will throw error to use deleteHoldings flag before proceed
```
    When user clicks confirm and their are related Holdings that do not reference any other pieces or items
    Then Delete Holdings confirmation is shown
    AND user can choose to Keep Holdings OR Delete Holding(s)
```

Reply:
I did not realize there was going to be two confirmation modals. OK.